tests/test_goldens.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2022 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+import difflib
+import filecmp
+import os
+import shutil
+import sys
+
+from docuploader import shell
+
+import pytest
+
+from sphinx.application import Sphinx
+
+
+@pytest.mark.parametrize(
+    "test_dir", ["handwritten", "gapic-combo", "gapic-auto"]
+)
+def test_goldens(update_goldens, test_dir):
+    source_dir = Path("tests/testdata") / test_dir
+    golden_dir = Path("tests/testdata/goldens") / test_dir
+
+    docs_dir = source_dir / "docs"
+    build_dir = docs_dir / "_build"
+
+    if build_dir.exists():
+        shutil.rmtree(build_dir)
+
+    out_dir = build_dir / "html/docfx_yaml"
+
+    # Install dependencies.
```

Comment:
This mimics how Sphinx works in the repositories as well as for the plugin repo. It must have the packages installed before generating documentation. I could put this in `tox.ini` or in `requirements.txt` I think, but it'd be test data specific and thought it'd be better to keep it generic based on the given package.

`Recommonmark` is required in order to run `sphinx-build` with the extension. It seems that it's required within docuploader's `shell` command, so either I needed to add it here or add it in docuploader. Since this is specific to this repository's usage I found it more appropriate to install it here locally for test usage.

Reply:
Any time you use `sphinx-docfx-yaml`, you'll need `Recommonmark`, right? So, it seems reasonable to include it as a dependency, even if it's only directly used by this repo for tests? I may be misunderstanding.

Do you have a link to how docuploader's `shell` requires it (what is it?)?

Reply:
Not quite - it's used for the extension but not directly in `sphinx-docfx-yaml`. That's why I included it as dependencies on each doc generation script (example: https://github.com/googleapis/python-logging/blob/77e621cb85b726a801227da85c31319f25969d19/noxfile.py#L312). When I tried including this in `requirements.txt` or `tox.ini` as part of dependency, `docuploader.shell` has trouble finding `recommonmark` required to run `sphinx-build` command and thus that's why I included it as a dependency.